scrapy_poet/repository.py

```diff
@@ -17,37 +17,27 @@
 providers = {}
 
 
-def register(provider_class: typing.Type[PageObjectInputProvider],
-             provided_class: typing.Optional[typing.Type] = None):
+def register(provider_class: typing.Type[PageObjectInputProvider]):
     """This method registers a Page Object Input Provider in the providers
-    registry. It could be replaced by the use of the ``provides`` decorator.
+    registry.
 
-    If ``provided_class`` is not specified, we get it from ``provider_class``.
-
-    Examples:
-
-        register(ResponseDataProvider)
-        register(ResponseDataProvider, ResponseData)
+    It could be replaced by the use of the ``provides`` decorator but you need
+    to make sure the module containing those classes are imported during
+    runtime.
     """
-    if provided_class is None:
-        provided_class = provider_class.provided_class
-
-    providers[provided_class] = provider_class
+    providers[provider_class.provided_class] = provider_class
```

Comment:
I would check here if `provided_class` is actually in `provider_class`. If not I would raise a meaningful exception.

Reply:
I've just added an assertion above this line. It should provide enough information about what's happening.